myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ExecutorCommandLineGenerator.java

```diff
@@ -19,11 +19,148 @@
+        cmdLine.append("/etc/hadoop/yarn-site.xml && ");
+      }
+    }
+  }
+
+  protected void addJavaOpt(StringBuilder opts, String propertyName, String propertyValue) {
+    String envOpt = String.format(PROPERTY_FORMAT, propertyName, propertyValue);
+    opts.append(envOpt);
+  }
+
+  protected void appendSudo(StringBuilder cmdLine) {
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      cmdLine.append(" sudo ");
+    }
+  }
+
+  protected void appendUserSudo(StringBuilder cmdLine) {
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      cmdLine.append(" sudo -E -u ");
+      cmdLine.append(myriadConfiguration.getFrameworkUser().get());
+      cmdLine.append(" -H ");
+    }
+  }
+
+  public String getConfigurationUrl() {
+    String httpPolicy = yarnConfiguration.get(TaskFactory.YARN_HTTP_POLICY);
+    String address = StringUtils.EMPTY;
+    if (httpPolicy != null && httpPolicy.equals(TaskFactory.YARN_HTTP_POLICY_HTTPS_ONLY)) {
+      address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_WEBAPP_HTTPS_ADDRESS);
+      if (StringUtils.isEmpty(address)) {
+        address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_HOSTNAME) + ":8090";
+      }
+    } else {
+      address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_WEBAPP_ADDRESS);
+      if (StringUtils.isEmpty(address)) {
+        address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_HOSTNAME) + ":8088";
+      }
+    }
+
+    return "http://" + address + "/yarnConfiguration";
+  }
+
+  private static String getFileName(String uri) {
+    int lastSlash = uri.lastIndexOf('/');
+    if (lastSlash == -1) {
+      return uri;
+    } else {
+      String fileName = uri.substring(lastSlash + 1);
+      Preconditions.checkArgument(!Strings.isNullOrEmpty(fileName), "URI should not have a slash at the end");
+      return fileName;
+    }
+  }
+
+  protected String getUser() {
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      return myriadConfiguration.getFrameworkSuperUser().get();
+    } else {
+      return myriadConfiguration.getFrameworkUser().get();
+    }
+  }
 
-  String getConfigurationUrl();
+  protected List<Protos.CommandInfo.URI> getUris() {
+    List<Protos.CommandInfo.URI> uris = new ArrayList<>();
+    if (myriadExecutorConfiguration.getJvmUri().isPresent()) {
+      final String jvmRemoteUri = myriadExecutorConfiguration.getJvmUri().get();
+      LOGGER.info("Getting JRE distribution from:" + jvmRemoteUri);
+      uris.add(Protos.CommandInfo.URI.newBuilder().setValue(jvmRemoteUri).build());
+    }
+    if (myriadExecutorConfiguration.getConfigUri().isPresent()) {
+      String configURI = myriadExecutorConfiguration.getConfigUri().get();
+      LOGGER.info("Getting Hadoop distribution from: {}", configURI);
+      uris.add(Protos.CommandInfo.URI.newBuilder().setValue(configURI).build());
+    }
+    if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
```

Comment:
Should the NodeManager URI be a required field? Seems like it should, but I may be missing something.